plugin-server/src/worker/ingestion/event-pipeline/2-emitToBufferStep.ts

```diff
@@ -17,6 +17,15 @@ export async function emitToBufferStep(
     ) => boolean = shouldSendEventToBuffer
 ): Promise<StepResult> {
     status.debug('🔁', 'Running emitToBufferStep', { event: event.event, distinct_id: event.distinct_id })
+
+    // Set `posthog_team.ingested_event` early such that e.g. the onboarding
+    // flow is allowed to proceed as soon as there has been an event starts
+    // processed as opposed to having to wait for the event to be buffered.
+    const team = await runner.hub.teamManager.fetchTeam(event.team_id)
+    if (team) {
+        await runner.hub.teamManager.setTeamIngestedEvent(team, event.properties || {})
+    }
+
```

Comment:
great for unblocking - maybe move it to line 37 though, so we only set if it went to the buffer, if it didn't go to the buffer the current flow is fine. 

Also let's re-review this once we have the live events. I'm a bit concerned about users having instrumented poorly show up as everything looks good

Reply:
Have moved!